packages/browser-pyright/src/browser-server.ts

```diff
@@ -285,3 +328,23 @@ export class PyrightServer extends LanguageServerBase {
         };
     }
 }
+
+export class BrowserBackgroundAnalysis extends BackgroundAnalysisBase {
+    constructor(console: ConsoleInterface) {
+        super(console);
+
+        const initialData: InitializationData = {
+            rootDirectory: (global as any).__rootDirectory as string,
+            cancellationFolderName: undefined,
+            runner: undefined,
+        };
+        const worker = createWorker(initialData);
+        this.setup(worker);
+    }
+}
+
+export class BrowserBackgroundAnalysisRunner extends BackgroundAnalysisRunnerBase {
+    constructor(initialData: InitializationData) {
+        super(parentPort(), initialData);
+    }
+}
```

Comment:
These are now very similar to the Node cases except for how the initial data is retrieved. Not sure it's worth unifying now but perhaps in future.